### PR TITLE
[Container] `az container create`: Add new parameters `--priority`, `--sku` and `--cce-policy` for container group

### PR DESCRIPTION
**Pull request by diyellap**
This PR:

Upgrades azure-mgmt-containerinstance to 10.1.0b1
Adds Priority property to Container Group. The Priority can be None, Regular or Spot.
Adds confidential SKU support for Container Group.
Tests priority property and confidential SKU.
[ContainerInstance] : `az container create': Added a new optional property "Priority" for Container Group creation.
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
container

**Comment by diyellap** [comment]
All container tests passed on local:
Capture

**Comment by diyellap** [comment]
[yonzhan](https://github.com/yonzhan) All container tests passed on my local. The CLI test pipeline is trying to rewrite the recording but is unable to because one of our tests is configured to record only once. That's why the Test Pipeline is failing here. Can you please take a look and unblock? Thanks.

**Comment by zhoxing-ms** [comment]
> E           AssertionError: Can't overwrite existing cassette ('/mnt/vss/_work/1/s/src/azure-cli/azure/cli/command_modules/container/tests/latest/recordings/test_container_create_with_acr.yaml') in your current record mode ('once').
E           No match for the request (<Request (PUT) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ContainerInstance/containerGroups/clicontainer000002?api-version=2022-10-01-preview>](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ContainerInstance/containerGroups/clicontainer000002?api-version=2022-10-01-preview%3E)) was found.
E           Found 1 similar requests with 1 different matcher(s) :
E           
E           1 - (<Request (PUT) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ContainerInstance/containerGroups/clicontainer000002?api-version=2021-09-01>).](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ContainerInstance/containerGroups/clicontainer000002?api-version=2021-09-01%3E).)
E           Matchers succeeded : ['method', 'scheme', 'host', 'port', 'path']
E           Matchers failed :
E           _custom_request_query_matcher - assertion failure :
E           None

In fact, this problem is caused by the upgrade of the SDK package `azure-mgmt-containerinstance`, which leads to the upgrade of api-version for `Microsoft.ContainerInstance` from `2021-09-01` to `2022-10-01-preview`. Please rerecord these tests in live mode, and then upload those yaml files

File: src/azure-cli/azure/cli/command_modules/container/_params.py [hunk 1]
```diff
@@ -74,6 +74,11 @@ def load_arguments(self, _):
         c.argument('secrets_mount_path', validator=validate_volume_mount_path, help="The path within the container where the secrets volume should be mounted. Must not contain colon ':'.")
         c.argument('file', options_list=['--file', '-f'], help="The path to the input file.")
         c.argument('zone', help="The zone to place the container group.")
+        c.argument('priority', help='The priority of the Container Group')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/container/_params.py:77]:
```suggestion
        c.argument('priority', help='The priority of the container group')
```

Comment (fnuarnav) [reply src/azure-cli/azure/cli/command_modules/container/_params.py]: updated

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**